packages/compiler-cli/src/ngtsc/typecheck/src/oob.ts

```diff
@@ -34,6 +36,19 @@ export interface OutOfBandDiagnosticRecorder {
    * @param ref the `TmplAstReference` which could not be matched to a directive.
    */
   missingReferenceTarget(templateId: string, ref: TmplAstReference): void;
+
+  /**
+   * Reports usage of a `| pipe` expression in the template for which the named pipe could not be
+   * found.
+   *
+   * @param templateId the template type-cehcking ID of the template which contains the unknown
+   * pipe.
+   * @param ast the `BindingPipe` invocation of the pipe which could not be found.
+   * @param sourceSpan the `AbsoluteSourceSpan` of the pipe invocation (ideally, this should be the
+   * source span of the pipe's name). This depends on the source span of the `BindingPipe` itself
+   * plus span of the larger expression context.
+   */
+  missingPipe(id: string, ast: BindingPipe, sourceSpan: AbsoluteSourceSpan): void;
```

Comment:
You seem to have missed the actual param name in the interface here:

```suggestion
  missingPipe(templateId: string, ast: BindingPipe, sourceSpan: AbsoluteSourceSpan): void;
```